Bottom: ea85575192f4187b0c501cc17c93bcc4135ebef2
Top:    6bbf20a6ae5748cf647adb9b72d51cceeabb0b10
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-28 12:47:44 +0100

more chopping


---

diff --git a/src/hid.h b/src/hid.h
index 8afd735..327c2b0 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -371,12 +371,6 @@ typedef enum
     void (*draw_rect) (hidGC gc_, Coord x1_, Coord y1_, Coord x2_, Coord y2_);
     void (*fill_circle) (hidGC gc_, Coord cx_, Coord cy_, Coord radius_);
     void (*fill_polygon) (hidGC gc_, int n_coords_, Coord *x_, Coord *y_);
-    void (*draw_polygon) (hidGC gc_, PolygonType *poly,
-                          const BoxType *clip_box);
-    void (*fill_pcb_pad) (hidGC gc_, PadType *pad, bool clip, bool mask);
-    void (*thindraw_pcb_pad) (hidGC gc_, PadType *pad, bool clip, bool mask);
-    void (*fill_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
-    void (*thindraw_pcb_pv) (hidGC fg_gc, hidGC bg_gc, PinType *pv, bool drawHole, bool mask);
     void (*fill_rect) (hidGC gc_, Coord x1_, Coord y1_, Coord x2_, Coord y2_);
 
 
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 47012b1..a324708 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -818,7 +818,7 @@ ghid_init_renderer (int *argc, char ***argv, GHidPort *port)
 
   /* Setup HID function pointers specific to the GL renderer*/
   ghid_hid.end_layer = ghid_end_layer;
-  ghid_hid.draw_polygon = ghid_draw_polygon;
+//  ghid_hid.draw_polygon = ghid_draw_polygon;
 }
 
 void
@@ -903,8 +903,8 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   /* If we don't have any stencil bits available,
      we can't use the hidgl polygon drawing routine */
   /* TODO: We could use the GLU tessellator though */
-  if (hidgl_stencil_bits() == 0)
-    ghid_hid.draw_polygon = common_draw_polygon;
+//  if (hidgl_stencil_bits() == 0)
+//    ghid_hid.draw_polygon = common_draw_polygon;
 
   glEnable (GL_BLEND);
   glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index ceb0058..b972b22 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1506,7 +1506,7 @@ void ps_ps_init (HID *hid)
   hid->draw_rect          = ps_draw_rect;
   hid->fill_circle        = ps_fill_circle;
   hid->fill_polygon       = ps_fill_polygon;
-  hid->draw_polygon       = ps_draw_polygon;
+//  hid->draw_polygon       = ps_draw_polygon;
   hid->fill_rect          = ps_fill_rect;
   hid->calibrate          = ps_calibrate;
   hid->set_crosshair      = ps_set_crosshair;
